services/movie_session.py

```diff
@@ -0,0 +1,49 @@
+from django.db.models import QuerySet
+
+from db.models import MovieSession
+
+
+def create_movie_session(
+        movie_show_time: str,
+        movie_id: int,
+        cinema_hall_id: int,
+) -> None:
+
+    MovieSession.objects.create(
+        show_time=movie_show_time,
+        cinema_hall_id=cinema_hall_id,
+        movie_id=movie_id
+    )
+
+
+def get_movies_sessions(session_date: str = None) -> QuerySet | MovieSession:
+    if session_date:
+        return MovieSession.objects.filter(show_time__date=session_date)
+    return MovieSession.objects.all()
+
+
+def get_movie_session_by_id(movie_session_id: int) -> MovieSession:
+    return MovieSession.objects.get(id=movie_session_id)
+
+
+def update_movie_session(
+        session_id: int,
+        show_time: str = None,
+        movie_id: int = None,
+        cinema_hall_id: int = None,
+) -> None:
+
+    movie_session = MovieSession.objects.filter(id=session_id)
+
+    if show_time:
+        movie_session.update(show_time=show_time)
+
+    if movie_id:
+        movie_session.update(movie_id=movie_id)
+
+    if cinema_hall_id:
+        movie_session.update(cinema_hall_id=cinema_hall_id)
+

```

Comment:
you can use .get() method to get the object, and then change the attributes of the object and then save(). So, in the result, you remove redundant queries